unit/java_bytecode/java_types/java_generic_symbol_type.cpp

```diff
@@ -0,0 +1,31 @@
+/*******************************************************************\
+
+ Module: Unit tests for java_types
+
+ Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+#include <testing-utils/catch.hpp>
+#include <java_types.h>
+
+SCENARIO("java_generic_symbol_type", "[core][java_types]")
+{
+  WHEN("MyType is LGenericClass<TX;TY;>;")
+  {
+    auto symbol_type = symbol_typet("MyType");
+    const auto generic_symbol_type = java_generic_symbol_typet(
+      symbol_type, "LGenericClass<TX;TY;>;", "PrefixClassName");
```

Comment:
Given the missing precondition in the constructor, as mentioned above, replace this with `"LMyType<TX;TY;>;"`.